--- heartbeat.module	2010-05-18 16:00:48.000000000 -0400
+++ heartbeat.new.module	2010-07-29 10:21:16.000000000 -0400
@@ -838,7 +838,7 @@ function heartbeat_heartbeat_message_inf
       'variables' =>
       array(
         '@username' => '[node:author-name-url]',
-        '@node_type' => '[node:type]',
+        '@node_type' => '[node:type-name]',
         '@node_title' => '[node:title-link]',
       ),
     ),
