diff --git includes/view.inc includes/view.inc
index d6cbee8..90c947b 100644
--- includes/view.inc
+++ includes/view.inc
@@ -608,6 +608,8 @@ class view extends views_db_object {
       $this->exposed_widgets = $exposed_form->render_exposed_form();
       if (form_set_error() || !empty($this->build_info['abort'])) {
         $this->built = TRUE;
+        // Don't execute the query, but rendering will still be executed to display the empty text.
+        $this->executed = TRUE;
         return empty($this->build_info['fail']);
       }
     }
