diff --git a/og_aggregator.module b/og_aggregator.module
index c98baea..01a27c6 100644
--- a/og_aggregator.module
+++ b/og_aggregator.module
@@ -187,14 +187,22 @@ function og_aggregator_block($op, $delta = 0, $edit = array()) {
         $result = db_query_range("SELECT i.* FROM {og_aggregator_item} i JOIN {og_aggregator_feed} f ON i.fid = f.fid WHERE f.nid = %d ORDER BY i.timestamp desc", $groupnode->nid, 0, variable_get('og_aggregator_block_items_number',10));
         $block['content'] = '<div class="more-link">'. 
 l(t('more'), "node/$groupnode->nid/aggregator/feed", array('title' => t('View this feed\'s recent news.'))) .'</div>';
+      
+        $items = array();
+        while ($item = db_fetch_object($result)) {
+          $items[] = theme('og_aggregator_block_item', $item);
+        }
+        
+        // If no items are available, don't return a block
+        if (empty($items)) {
+          return NULL;
+        }
+        
+        $block['content'] = theme('item_list', $items) . $block['content'];
+     
+        return $block;
       }
-      $items = array();
-      while ($item = db_fetch_object($result)) {
-        $items[] = theme('og_aggregator_block_item', $item);
-      }
-      $block['content'] = theme('item_list', $items) . $block['content'];
     }
-    return $block;
   }
 }
 
