Index: userpoints.module
===================================================================
RCS file: /cvs/drupal-contrib/contributions/modules/userpoints/userpoints.module,v
retrieving revision 1.67.2.48
diff -u -r1.67.2.48 userpoints.module
--- userpoints.module	21 Oct 2009 17:55:23 -0000	1.67.2.48
+++ userpoints.module	1 Dec 2009 10:38:36 -0000
@@ -1672,9 +1672,9 @@
   // User which are displaying points for
   $uid = (int)arg(1);
   if ($uid) {
-    if ($uid != $user->uid) {
+    if ($uid != $user->uid && !user_access(USERPOINTS_PERM_ADMIN)) {
       // If not the same as logged in user
-		  drupal_access_denied();
+      drupal_access_denied();
       return;
     }
   }
@@ -1685,17 +1685,11 @@
   $point_user = user_load(array('uid' => $uid));
 
   // Check permissions
-  if(!user_access(USERPOINTS_PERM_VIEW)) {
+  if(!user_access(USERPOINTS_PERM_VIEW) && !user_access(USERPOINTS_PERM_VIEW_OWN)) {
     // User has no access to view points
-		drupal_access_denied();
+    drupal_access_denied();
     return;
   }
-
-  if (!user_access(USERPOINTS_PERM_VIEW_OWN)) {
-    // User has no access to view own points
-		drupal_access_denied();
-    return;
-  } 
 		
   $title = t('!Points for ', userpoints_translation()) . check_plain($point_user->name);
   drupal_set_title($title);

